pkg/vfs/vfs_unix.go

```diff
@@ -290,3 +291,30 @@ func (v *VFS) Flock(ctx Context, ino Ino, fh uint64, owner uint64, typ uint32, b
 	}
 	return
 }
+
+const (
+	FS_IOC_GETFLAGS = 0x80086601
+	FS_IOC_SETFLAGS = 0x40086602
+)
+
+func (v *VFS) Ioctl(ctx Context, ino Ino, cmd uint32, arg uint64, data []byte) (err syscall.Errno) {
+	defer func() { logit(ctx, "ioctl (%d,0x%X,0x%X,%v): %s", ino, cmd, arg, data, strerr(err)) }()
+	if cmd != FS_IOC_GETFLAGS && cmd != FS_IOC_SETFLAGS {
+		err = syscall.ENOTTY
+		return
+	}
+	if IsSpecialNode(ino) {
+		err = syscall.EPERM
+		return
+	}
+	var attr = &Attr{}
+	if cmd == FS_IOC_GETFLAGS {
+		err = v.Meta.GetAttr(ctx, ino, attr)
+		utils.NativeEndian.PutUint64(data, uint64(attr.Flags))
+		return
+	}
+	iflag := utils.NativeEndian.Uint64(data)
+	attr.Flags = uint8(iflag)
```

Comment:
There could be some flags greater than 256, and these flags could be different across OS, so we need to mapping these flags into internal ones (less than 256).

Reply:
Juicefs appears to support Linux, Windows, and MacOS. I can't find out how Windows and MacOS do these things. For MacOS, there is a chflags command, but I don't have a MaC to test it :smiley: . And the FS_IOC_GETFLAGS/FS_IOC_SETFLAGS seems to be Linux-specific.

Reply:
We may want to support these flags in macOS as well, see http://hints.macworld.com/article.php?story=20031017061722471

Btw, even in Linux, the flags can be wider than 8bits, we should do the mapping (from 32/64bits to 8bits).

So, so we can start with the ioctl callback, then add some flags one by one.

Reply:
> Btw, even in Linux, the flags can be wider than 8bits, we should do the mapping (from 32/64bits to 8bits).
>
The flags is defined [here](https://elixir.bootlin.com/linux/v5.19.2/source/include/uapi/linux/fs.h#L239), the flags we want to support is fit in low 8 bit, do we need the mapping.
 
> So, so we can start with the ioctl callback, then add some flags one by one.
>
Sorry, I don't get the point.

Reply:
We may want to support other flags which is greater than 256, so it's better to have the mapping.

Reply:
Done the mapping, please take a look.